src/ruby/lib/grpc/generic/service.rb

```diff
@@ -167,22 +167,22 @@ def initialize(host, creds, **kw)
             if desc.request_response?
               define_method(mth_name) do |req, metadata = {}|
                 GRPC.logger.debug("calling #{@host}:#{route}")
-                request_response(route, req, marshal, unmarshal, metadata)
```

Comment:
I think I see the issue - renaming the metadata variable here to `client_stub_opts` would be more clear?